File: python/src/commands/exists.py
from .base import BaseCommand
from src.resp.serializer import RespSerializer

class Exists(BaseCommand):
  @classmethod
  def command(cls) -> bytes:
    return b'exists'
  
  def run(self) -> bytes:
    exists: int = 0
    store_keys: list[bytes] = self.store.keys()
    for arg in self.args:
      exists += 1 if arg in store_keys else 0
    
    return RespSerializer.serialize(exists)
  

File: ruby/lib/redis/commands/exists.rb
require_relative './base'

module Redis
  module Commands
    class Exists < Base
      def self.command
        'exists'
      end
      
      def run
        exist_count = 0
        args.each do |arg|
          if @store[arg]
            exist_count += 1
          end
        end

        Redis::Resp::Serializer.serialize(exist_count)
      end 
    end
  end
end